File: examples/chill/jacobi_1d.f
      SUBROUTINE jacobi(N,A,B)
      INTEGER N,T,I

      REAL*8 A(N),B(N)

      DO 10, T=1,100
         DO 20, I=2,N-1
            B(I)=0.25D0*(A(I-1)+A(I+1))+0.5D0*A(I)
 20      CONTINUE
         DO 30, I=2,N-1
            A(I) = B(I)
 30      CONTINUE
 10   CONTINUE

      END
      

File: examples/chill/testcases/jacobi_1d.c

#define N       1024


void jacobi(double A[N], double B[N]) {
    int t,i;
    for(t = 1; t < N; t++) {
        for(i = 1; i < N-1; i++) {
            B[i] = 0.25*(A[i-1] + A[i+1]) + 0.5*A[i];
        }
        
        for(i = 1; i < N-1; i++) {
            A[i] = B[i];
        }
    }
}


